### ISSUE TITLE
`repr` and `renderTree` don't work with generic object containing `nkRecWhen`

### ISSUE THREAD

**Issue by timotheecour** [nim-lang/Nim#14708]
Found this while investigating https://github.com/nim-lang/Nim/issues/14698: `repr` doesn't work with generic object containing `nkRecWhen`

### Example
```nim
when true:
  macros
  type Foo[T] = object
    x1: int
    when false: discard
    when true: x2: float
  macro bar(a: typed) =
    var a = a[0].getImpl
    echo a.repr
    # echo a.treeRepr
  bar(Foo[int])
```

### Current Output
```
Foo[T] = object
  x1: int
  when false:

  when true:
    x2
```

### Expected Output
```
Foo[T] = object
  x1: int
  when false: discard
  when true:
    x2: float
```

### Additional Information
devel 1.3.5 ff93302f69c724bc03d7e0529fb0b9b6bfa0b36e

using `renderTree` inside compiler code has the same problem.

**Comment by metagn** [comment]
Now gives

```
Foo[T] = object
  x1: int
  when false:
    nil
  when true:
    x2: float
```

There's probably already a test case for repr for generic object with when, just add `when false: discard` to it

**Comment by metagn** [comment]
Fixed by #21017, related #9899
